packages/client/src/applications/index.ts

```diff
@@ -623,6 +651,30 @@ async function updateFieldAgentDeclaredApplicationsByUser(
     declaredApplications.results
   )
 
+  currentUserData = {
+    ...currentUserData,
+    applications: currentUserData.applications.filter(
+      application =>
+        !rejectedApplicationId.includes(application.compositionId as string)
+    )
+  }
+
+  allUserData = allUserData.map(userData => {
+    if (userData.userID !== currentUserData!.userID) {
+      return {
+        ...userData
+      }
+    } else {
+      return {
+        ...userData,
```

Comment:
No need to do the same calculation twice as the `currentUserData` can be returned from here.